qt/scientific_interfaces/ISISReflectometry/Common/ValidationResult.h

```diff
@@ -0,0 +1,74 @@
+// Mantid Repository : https://github.com/mantidproject/mantid
+//
+// Copyright &copy; 2018 ISIS Rutherford Appleton Laboratory UKRI,
+//     NScD Oak Ridge National Laboratory, European Spallation Source
+//     & Institut Laue - Langevin
+// SPDX - License - Identifier: GPL - 3.0 +
+#ifndef MANTID_ISISREFLECTOMETRY_RESULT_H
+#define MANTID_ISISREFLECTOMETRY_RESULT_H
+#include <boost/optional.hpp>
+#include <boost/variant.hpp>
+
+namespace MantidQt {
+namespace CustomInterfaces {
+
+template <typename Validated, typename Error = boost::blank>
+class ValidationResult {
```

Comment:
This looks something like the expectation c++ proposal, i'd appreciate some pointers to where this is really needed/used.